### PR TITLE
[libdef] prismic.io v3.x.x

### PR DESCRIPTION
**Pull request by ctrlplusb**
_Caveat_: I decided to support the latest promise API.  We can extend the definitions to include the optional callback mechanism if there is demand for it.

### CODE REVIEW

**Comment by ctrlplusb** [comment]
Resolves #179

File: definitions/npm/prismic.io_v3.x.x/flow_>=v0.28.x/prismic.io_v3.x.x.js [hunk 1]
```diff
@@ -0,0 +1,331 @@
+declare module 'prismic.io' {
+  declare type linkResolver = (...args: any) => any;
+
+  declare type DaysOfWeek = 'Monday'
```
Comment (jeffmo) [inline definitions/npm/prismic.io_v3.x.x/flow_>=v0.28.x/prismic.io_v3.x.x.js]: ocd protip: You can have leading `|`s in unions:

```js
declare type DaysOfWeek = 
  | 'Monday'
  | 'Tuesday'
  | 'Wednesday'
  ...
```

You don't have to change this, but figured I'd mention it incase you like it :)

File: definitions/npm/prismic.io_v3.x.x/test_prismic.io_v3.x.x.js [hunk 2]
```diff
@@ -0,0 +1,281 @@
+// @flow
```
Comment (jeffmo) [inline definitions/npm/prismic.io_v3.x.x/test_prismic.io_v3.x.x.js]: Can we [add a few `$ExpectError`](https://github.com/flowtype/flow-typed/blob/master/CONTRIBUTING.md#4-write-a-test-file-whose-name-starts-with-test_-in-the-package-version-directory) tests in here as well to sanity check that we get errors when we're supposed to?

**Comment by ctrlplusb** [comment]
Thanks for the tips.  I've done this.  I see the build is failing, but it's due to an issue with i18next definitions.

**Comment by jeffmo** [comment]
I just pushed https://github.com/flowtype/flow-typed/commit/9141ae363c91b2c2e258e1e883aecbc1321dec1c if you want to rebase on top of it to get a clean build

**Comment by jeffmo** [comment]
(otherwise we can just verify with the build results -- should auto-merge cleanly anyway)

**Comment by ctrlplusb** [comment]
Please don't be afraid to be as OCD as you like with reviewing my definition.  I appreciate any type of corrections at all as I plan to be writing lots more definitions. :)

**Comment by jeffmo** [comment]
awesome!

**Pull request merged**